src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs

```diff
@@ -186,8 +164,8 @@ private void CompileXsltToQil(object stylesheet, XsltSettings settings, XmlResol
 
         private void CompileQilToMsil(XsltSettings settings)
         {
-            _command = new XmlILGenerator().Generate(_qil!, /*typeBuilder:*/null)!;
-            _outputSettings = _command.StaticData.DefaultWriterSettings;
+            _command = new XmlILGenerator().Generate(_qil!, null)!;
```

Comment:
I know this isn't your change but that trailing `!` looks weird. Do you think there's a way to annotate the Generate call with nullable attributes so it isn't needed?

Reply:
@Wraith2 Unfortunately, there is no attribute in the annotations for this. [return: NotNullIfNull(memberName)]

Reply:
Hmm, sharplab seems to suggest `[return: System.Diagnostics.CodeAnalysis.NotNull]` would work even with the return type being nullable. https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMAGAsAKAGIYBXAGzImDIFMACGmK2ggrARgKbBoGcAHCAGN6ABQIBvAnRl0sAZmmyp+WWrmZMdJerordu9O2x0IdALx0YNAO50AGgAoAlADoA4oxoAnCABcaFwBuHQMZIwBORwhXABlGAHM/AAtnENUDAF9Q7IyZUPR5DQdQ/QMAbXQAdhA5dlcAESgIBJgAe14/KCFeVwBhNoQaAEEmMgBPXihegDk2vxnyMgBdULVCuuwAfjpPa18AlzXlY8MqugBlP29YBNcAUTB+P3H0sNy1XMygA

Reply:
@Wraith2 Is it a little trickier if TypeBuilder? typeBldr is null, then notnull should be returned

Reply:
https://github.com/dotnet/runtime/issues/59519

Reply:
Hmm, if it's nuanced then I'm not sure I'd have gone with the original unconditional trailing `!` because to me that's a blanket assertion that it will not be null. I'll stop poking at it and leave you to it.